projects/packages/my-jetpack/_inc/data/welcome-banner/use-welcome-banner.ts

```diff
@@ -22,8 +25,19 @@ const useWelcomeBanner = () => {
 		),
 	} );
 
+	/**
+	 * Dismiss the welcome banner and remove it from the initial state.
+	 * Using a separate function instead of onSuccess so the frontend picks up the
+	 * change before the mutation is complete.
+	 */
+	const dismissWelcomeBannerAndRemoveFromState = () => {
+		delete redBubbleAlerts[ 'welcome-banner-active' ];
+		setMyJetpackRedBubbleAlerts( redBubbleAlerts );
```

Comment:
I think we have some redundancy in how the banner visibility is controlled.

I think this bit of state in the welcome banner component is actually what's hiding the banner when it's dismissed:
```
const [ bannerVisible, setBannerVisible ] = useState( isWelcomeBannerVisible );
```

So we might not need this here?